lib/Parse/ParseExpr.cpp

```diff
@@ -3115,10 +3122,12 @@ ExprResult Parser::ParseBoundsCastExpression(IdentifierInfo &Ident,
   // Match the ')'.
   T.consumeClose();
   RParenLoc = T.getCloseLocation();
-  if (!E1.isInvalid())
-    Result = Actions.ActOnBoundsCastExpr(getCurScope(), ILoc, Ty.get(),
-                                         RelativeClause, RParenLoc, E1.get(),
-                                         E2.get(), E3.get(), kind);
+
+  Result = Actions.ActOnBoundsCastExpr(
+      getCurScope(), OpLoc, Kind, LAngleBracketLoc, Ty.get(), RAngleBracketLoc,
+      RelativeClause, LParenLoc, RParenLoc, E1.get(), E2.get(), E3.get(),
+      syntax);
+
```

Comment:
I think if there is an error about E1, E2, E3 and TypeResult, then the code return ExprError(). code is not reached at this point if one of them is invalid. If want me to check them again, I will modify the code to check again here.